### PR TITLE
FOUNDATIONS: Profile Retrieve All

### PR DESCRIPTION
**Pull request by rcruzfreelance**

### CODE REVIEW

File: Taarafo.Core/Services/Foundations/Profiles/ProfileService.cs [hunk 1]
```diff
@@ -29,11 +29,18 @@ public partial class ProfileService : IProfileService
...
             ValidateProfileOnAdd(profile);
 
             return await this.storageBroker.InsertProfileAsync(profile);
         });
+
+        public IQueryable<Profile> RetrieveAllProfiles() =>
+        TryCatch(() =>
+        {
+            return this.storageBroker.SelectAllProfiles();
+
```
Comment (hassanhabib) [inline Taarafo.Core/Services/Foundations/Profiles/ProfileService.cs:43]: Take this line out. And one-line this whole statement please

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RetrieveAll.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectAllProfiles());
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogCritical(It.Is(SameExceptionAs(
+                    expectedProfileDependencyException))),
+                        Times.Once);
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
```
Comment (hassanhabib) [inline Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RetrieveAll.cs:50]: `dateTimeBroker` validates are missing here.

File: Taarafo.Core/Services/Foundations/Profiles/ProfileService.cs (hunk 1 above)
Comment (rcruzfreelance) [reply Taarafo.Core/Services/Foundations/Profiles/ProfileService.cs]: done

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RetrieveAll.cs (hunk 2 above)
Comment (rcruzfreelance) [reply Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RetrieveAll.cs:49]: dateTimeBroker was added for validation

**Pull request merged**